services/gitlab/gitlab-coverage.service.js

```diff
@@ -0,0 +1,143 @@
+</p>
+<p>
+  Your badge should be working fine now.
+</p>
+`
+
+module.exports = class GitlabCoverage extends BaseSvgScrapingService {
+  static get category() {
+    return 'coverage'
+  }
+
+  static get route() {
+    return {
+      base: 'gitlab/coverage',
+      pattern: ':user/:repo/:branch',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Gitlab code coverage',
+        namedParams: {
+          user: 'gitlab-org',
+          repo: 'gitlab-runner',
+          branch: 'master',
+        },
+        staticPreview: this.render({ coverage: 67 }),
+        documentation,
+      },
+      {
+        title: 'Gitlab code coverage (specific job)',
+        namedParams: {
+          user: 'gitlab-org',
+          repo: 'gitlab',
+          branch: 'master',
+        },
+        queryParams: { job_name: 'rspec:coverage' },
+        staticPreview: this.render({ coverage: 96 }),
+        documentation,
+      },
+      {
+        title: 'Gitlab code coverage (self-hosted)',
+        namedParams: { user: 'GNOME', repo: 'libhandy', branch: 'master' },
+        queryParams: { gitlab_url: 'https://gitlab.gnome.org' },
+        staticPreview: this.render({ coverage: 93 }),
+        documentation,
+      },
+      {
+        title: 'Gitlab code coverage (self-hosted, specific job)',
+        namedParams: { user: 'GNOME', repo: 'libhandy', branch: 'master' },
+        queryParams: {
+          gitlab_url: 'https://gitlab.gnome.org',
+          job_name: 'unit-test',
+        },
+        staticPreview: this.render({ coverage: 93 }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'coverage' }
+  }
+
+  static render({ coverage }) {
+    return {
+      message: `${coverage.toFixed(0)}%`,
+      color: coveragePercentage(coverage),
+    }
+  }
+
+  async fetch(
+    { user, repo, branch },
+    { gitlab_url: baseUrl = 'https://gitlab.com', job_name }
+  ) {
+    // Since the URLdoesn't return a usable value when an invalid job name is specified,
+    // it is recommended to not use the query param at all if not required
+    job_name = job_name ? `?job=${job_name}` : ''
+    const url = `${baseUrl}/${user}/${repo}/badges/${branch}/coverage.svg${job_name}`
+    const errorMessages = {
+      401: 'repo not found',
+      404: 'repo not found',
+    }
+    return this._requestSvg({
+      schema,
+      url,
+      errorMessages,
+    })
+  }
+
+  static transform({ coverage }) {
+    if (coverage === 'unknown') {
+      throw new NotFound({ prettyMessage: 'not set up' })
+    }
+    return this.render({
+      coverage: Number(coverage.slice(0, -1)),
+    })
+  }
+
+  async handle({ user, repo, branch }, { gitlab_url, job_name }) {
+    const svg = await this.fetch(
+      { user, repo, branch },
+      { gitlab_url, job_name }
+    )
+    return this.constructor.transform({ coverage: svg.message })
```

Comment:
Our handle function still needs to call render, so this could either be split into two lines

```suggestion
    const { coverage } = this.constructor.transform({ coverage })
    return this.constructor.render({ coverage })
```

or oneline:
or one line
```suggestion
    return this.constructor.render({ coverage: this.constructor.transform({ coverage }) })
```